e2-studio-star-rx72n-firmware/tests/test_rx_frame.c

```
@@ -2240,6 +2240,235 @@ void test_roundtrip_reset(void)
+  uint8_t    misaligned[k_frame_max_size + k_resync_prefix_len];
+  uint32_t   wire_len = 0;
+
+  /* Build a simple ACK frame */
+  TEST_ASSERT_EQUAL(k_rx_ok, rx_frame_create_ack(&frame, k_test_seq_one));
+  TEST_ASSERT_EQUAL(k_rx_ok, rx_frame_encode(&s_encoder, &frame, wire, &wire_len));
+
+  /* Prepend one junk byte to simulate a dropped byte desynchronizing the stream */
+  misaligned[k_resync_zero] = k_resync_junk_byte;
+  memcpy(&misaligned[k_resync_prefix_len], wire, wire_len);
+
+  uint32_t discarded = k_resync_zero;
```

Clang (14.0.6)

[warning] variable 'discarded' is not initialized

(cppcoreguidelines-init-variables)